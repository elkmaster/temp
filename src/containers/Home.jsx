import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { fetchWeather } from '../actions/weather';

import '../styles/components/home.css';

export class Home extends Component {
  static propTypes = {
    data: PropTypes.shape({
      city: PropTypes.string,
      temp: PropTypes.number,
    }).isRequired,
    fetchWeather: PropTypes.func.isRequired,
  };

  componentDidMount() {
    this.props.fetchWeather();
  }

  render() {
    const { data } = this.props;

    return (
      <div className="container">
        <p className="description">This is Home component.</p>
        <p className="description">Below a real API request with Cherkasy weather.</p>
        <p className="weather-data">
          Temperature in <span className="city">{data.city}</span> - <span className="temp">{data.temp}</span>
        </p>
      </div>
    );
  }
}

const mapStateToProps = ({ weather: { data } }) => ({ data });

export default connect(mapStateToProps, { fetchWeather })(Home);
