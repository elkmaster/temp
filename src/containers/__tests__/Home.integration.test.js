import React from 'react';
import { mount } from 'enzyme';
import MockAdapter from 'axios-mock-adapter';
import flushPromises from '../../utils/testHelpers';
import createNormalStore from '../../utils/testHelpers/normalStore';

import Home from '../Home';
import axios from 'axios/index';

const mock = new MockAdapter(axios);

const setup = () => {
  const store = createNormalStore();
  const wrapper = mount(
    <Home store={store} />,
  );

  return {
    wrapper,
    store,
  };
};

describe('Home container', () => {
  const url = `http://api.openweathermap.org/data/2.5/weather?q=Cherkasy&units=metric&appid=${process.env.REACT_APP_WEATHER_API_KEY}`;
  mock.onGet(url).reply(200, {
    name: 'Cherkasy',
    main: {
      temp: 22,
    },
  });

  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call `componentDidMount` once', () => {
    const spy = jest.spyOn(Home.prototype, 'componentDidMount');
    setup();
    expect(spy).toHaveBeenCalledTimes(1);
    spy.mockRestore();
  });

  it('should fetch data on mount', async () => {
    const { store } = setup();
    await flushPromises();
    expect(store.getState().weather.data).toEqual({
      city: 'Cherkasy',
      temp: 22,
    });
  });

  it('should display fetched data', async () => {
    const { wrapper } = setup();
    await flushPromises();
    wrapper.update();
    expect(wrapper.find('.temp').text()).toBe('22');
    expect(wrapper.find('.city').text()).toBe('Cherkasy');
  });
});


