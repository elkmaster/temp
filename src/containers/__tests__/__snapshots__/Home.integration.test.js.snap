// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home container should render correctly 1`] = `
<Connect(Home)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Home
    data={Object {}}
    fetchWeather={[Function]}
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="container"
    >
      <p
        className="description"
      >
        This is Home component.
      </p>
      <p
        className="description"
      >
        Below a real API request with Cherkasy weather.
      </p>
      <p
        className="weather-data"
      >
        Temperature in 
        <span
          className="city"
        />
         - 
        <span
          className="temp"
        />
      </p>
    </div>
  </Home>
</Connect(Home)>
`;
